### ISSUE TITLE
Add controls to Storybook embeds

### ISSUE THREAD

**Issue by mattrosno** [carbon-design-system/carbon-platform#1443]
**Summary**

> What problem (or opportunity) are we solving? Why are we solving it? Is there supporting user
> conversations or research?

The legacy live demo is effective because there are initial variants in the selector, and then a right panel that shows the most important properties and their values. This allows you to see any of the component's variants - not only what's available in the dropdown.

<img width="960" alt="image" src="https://user-images.githubusercontent.com/1691245/196727891-76b80271-377a-4887-b5f4-11e351b91e40.png">

The legacy live demo only supports React components, and is error-prone, so the new direction used in Carbon Platform is to use Storybook embeds.

<img width="642" alt="image" src="https://user-images.githubusercontent.com/1691245/196728825-3fee3bcd-c60e-4836-8901-0592d62bb628.png">

In doing so, you can select a variant which is just a Storybook story, and then you can view the full demo to then modify controls from within Storybook, not Carbon's website.

**Job stories**

> What value will this deliver?

1. When learning the capabilities of a component, I want to see all available properties and values, so I can have confidence in what the component can and can't do without needing to read its source code.

**Acceptance criteria**

> What requirements must be met to complete this request?

1. To do

**Comment by mattrosno** [comment]
More info coming soon!

**Comment by mattrosno** [comment]
August 18th update on Storybook 7.0: https://storybook.js.org/blog/7-0-design-alpha

>**Docs is getting overhauled in 7.0** thanks to community feedback. There's not enough space in this post to cover all the improvements. Sign up to the mailing list to get a sneak peek when it's announced (MDX2, **embed**, and much more).

What does "embed" mean here? The "Docs" tab moves to the side bar. Will there also be improvements to how documentation can be embedded (e.g. including controls)?

---

I couldn't find any related issues, open or closed. This request still "needs triaged": https://github.com/storybookjs/storybook/issues/17117

---

The 7.0 alphas are available: https://www.npmjs.com/package/storybook

Looking through the `next` changelog... I don't see anything about controls being added to embeds: https://github.com/storybookjs/storybook/blob/next/CHANGELOG.md

---

TL;DR: I don't see anything (via issues, discussions, changelogs, next branch) that indicates that embeddable controls are coming.

**Comment by mattrosno** [comment]
Per https://storybook.js.org/docs/react/writing-docs/doc-block-argstable, this is cool, we can group the controls:

![image](https://user-images.githubusercontent.com/1691245/196794257-1002de90-3faa-414a-84f0-ae7bee4ff0ad.png)

A problem with the current Stories is button, for example, has 18 controls: https://react.carbondesignsystem.com/?path=/story/components-button--default

That's great for development, but for designers looking to explore component capability, it'd be nice to group the controls (properties and their values) that are most important, like `disabled`, `hasIconOnly`, `isExpressive`, `kind`, `size`, etc. and not controls like `onMouseLeave`.

---

Another option, if we want to have a story with just the most important controls shown, you can disable controls for specific properties: https://storybook.js.org/docs/react/essentials/controls#disable-controls-for-specific-properties

That could reduce the number of Button controls from 18 to ~5.
